qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,149 @@
+import logging
+
+from virttest import error_context
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Assign cpu to numa node with "-numa cpu", check the numa info in monitor
+    and guest os match with the qemu cli
+    """
+
+    def convert_cpu_topology_to_index(socket, die, core, thread):
+        """
+        Convert the cpu topology to cpu index list
+        """
+        def _get_boundary(value, max_value, weight):
+            """
+            Get the data range of one bit
+
+            :param value: the current value of the bit
+            :param max_value: the max value of the bit
+            :param weight: the weight of the bit
+            """
+            min_boundary = int(value if value is not None else 0) * weight
+            max_boundary = int(value if value is not None else (max_value - 1)) * weight
+            return (min_boundary, max_boundary)
+
+        if vm_arch in ('x86_64', 'i386'):
+            socket_min, socket_max = _get_boundary(socket, vcpu_sockets, socket_weight)
+            die_min, die_max = _get_boundary(die, vcpu_dies, die_weight)
+            core_min, core_max = _get_boundary(core, vcpu_cores, core_weight)
+            thread_min, thread_max = _get_boundary(thread, vcpu_threads, thread_weight)
+            cpu_min = socket_min + die_min + core_min + thread_min
+            cpu_max = socket_max + die_max + core_max + thread_max
+        elif vm_arch in ('ppc64', 'ppc64le'):
+            cpu_min = int(core)
+            cpu_max = int(core) + vcpu_threads - 1
+        cpu_index = list(range(cpu_min, cpu_max + 1))
+        return cpu_index
```

Comment:
Let's use cpu_list, cpu_index is kind of confusing.

Reply:
> Let's use cpu_list, cpu_index is kind of confusing.

Updated, thanks!